oca/api.py

```diff
@@ -49,15 +32,66 @@ def method(self):
 
             [1, 2, 3]
 
+        If no container is provided as an argument to the decorator, and a
+        value is returned from the decorated method, a ValueError will be
+        raised.
+
+        .. code-block:: python
+
+            @api.foreach
+            def method(self):
+                # Raises ValueError
+                return 1+2
+
+            @api.foreach
```

Comment:
Assuming that `@api.foreach` is equivalent to `@api.foreach()` is a little bit confusing. I think I prefer that you don't support the `@api.foreach` way, so we always have to call it. Similar to [`@http.route()`](https://github.com/odoo/odoo/blob/10.0/odoo/http.py#L417).